<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>931</width>
    <height>905</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Fuzzy Logic Builder</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>fuzzy.bmp</normaloff>fuzzy.bmp</iconset>
  </property>
  <property name="layoutDirection">
   <enum>Qt::LeftToRight</enum>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QWidget" name="formLayoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>400</y>
      <width>911</width>
      <height>441</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout" columnstretch="0">
     <property name="leftMargin">
      <number>9</number>
     </property>
     <property name="topMargin">
      <number>9</number>
     </property>
     <property name="rightMargin">
      <number>9</number>
     </property>
     <property name="bottomMargin">
      <number>9</number>
     </property>
     <item row="1" column="0">
      <widget class="GraphWidget" name="graphicsView"/>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="horizontalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>0</y>
      <width>791</width>
      <height>29</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_11">
     <item>
      <widget class="QPushButton" name="AddLogic">
       <property name="maximumSize">
        <size>
         <width>8000</width>
         <height>40</height>
        </size>
       </property>
       <property name="text">
        <string>Add Logic</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="DelLogic">
       <property name="text">
        <string>Del Logic</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="EditLogic">
       <property name="text">
        <string>Edit</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="Group">
       <property name="maximumSize">
        <size>
         <width>200</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="text">
        <string>Configure Groups</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="GroupSelect">
       <property name="maximumSize">
        <size>
         <width>200</width>
         <height>16777215</height>
        </size>
       </property>
       <item>
        <property name="text">
         <string>Show All</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="Lock">
       <property name="maximumSize">
        <size>
         <width>80</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="text">
        <string>Locked</string>
       </property>
       <property name="checked">
        <bool>false</bool>
       </property>
       <property name="tristate">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="DeleteBlock">
       <property name="maximumSize">
        <size>
         <width>40</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="autoFillBackground">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>X</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="LoadTest">
       <property name="text">
        <string>Load test.c</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>30</y>
      <width>921</width>
      <height>361</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="ValueTab">
     <attribute name="title">
      <string>Values</string>
     </attribute>
     <widget class="QWidget" name="verticalLayoutWidget_3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>901</width>
        <height>314</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <property name="topMargin">
        <number>0</number>
       </property>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_10">
         <property name="topMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="LogicLabel">
           <property name="text">
            <string>Name</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="NodeName"/>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_2" stretch="0,0,0,0,0,0,80">
           <property name="spacing">
            <number>6</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_8">
             <item>
              <widget class="QLineEdit" name="MinText">
               <property name="maximumSize">
                <size>
                 <width>80</width>
                 <height>16777215</height>
                </size>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="MinLabel">
               <property name="text">
                <string>Min</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="MinScaleLabel">
               <property name="maximumSize">
                <size>
                 <width>80</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>Scale</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDial" name="MinScale">
               <property name="maximumSize">
                <size>
                 <width>80</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="maximum">
                <number>512</number>
               </property>
               <property name="wrapping">
                <bool>false</bool>
               </property>
               <property name="notchTarget">
                <double>16.000000000000000</double>
               </property>
               <property name="notchesVisible">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_7">
             <item>
              <widget class="QSlider" name="Min">
               <property name="maximum">
                <number>1024</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="tickPosition">
                <enum>QSlider::TicksBothSides</enum>
               </property>
               <property name="tickInterval">
                <number>64</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_6">
             <item>
              <widget class="QLineEdit" name="MaxText">
               <property name="maximumSize">
                <size>
                 <width>80</width>
                 <height>16777215</height>
                </size>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="MaxLabel">
               <property name="text">
                <string>Max</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="MaxScaleLabel">
               <property name="maximumSize">
                <size>
                 <width>80</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>Scale</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDial" name="MaxScale">
               <property name="maximumSize">
                <size>
                 <width>80</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="maximum">
                <number>512</number>
               </property>
               <property name="pageStep">
                <number>10</number>
               </property>
               <property name="notchTarget">
                <double>16.000000000000000</double>
               </property>
               <property name="notchesVisible">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_5">
             <item>
              <widget class="QSlider" name="Max">
               <property name="maximum">
                <number>1024</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="tickPosition">
                <enum>QSlider::TicksBothSides</enum>
               </property>
               <property name="tickInterval">
                <number>64</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <item>
              <widget class="QLabel" name="pidlabel">
               <property name="text">
                <string>d</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <item>
              <widget class="QSlider" name="pid">
               <property name="maximum">
                <number>1024</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="invertedAppearance">
                <bool>false</bool>
               </property>
               <property name="invertedControls">
                <bool>false</bool>
               </property>
               <property name="tickPosition">
                <enum>QSlider::TicksBothSides</enum>
               </property>
               <property name="tickInterval">
                <number>64</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <property name="topMargin">
              <number>7</number>
             </property>
             <item>
              <widget class="QCustomPlot" name="Graph" native="true"/>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QPushButton" name="SetPoint">
             <property name="text">
              <string>Set Data Point</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_9">
             <property name="spacing">
              <number>6</number>
             </property>
             <property name="leftMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QTableView" name="ValueTable">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="frameShape">
                <enum>QFrame::WinPanel</enum>
               </property>
               <property name="horizontalScrollBarPolicy">
                <enum>Qt::ScrollBarAlwaysOff</enum>
               </property>
               <attribute name="horizontalHeaderDefaultSectionSize">
                <number>90</number>
               </attribute>
               <attribute name="horizontalHeaderMinimumSectionSize">
                <number>80</number>
               </attribute>
               <attribute name="verticalHeaderVisible">
                <bool>false</bool>
               </attribute>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="Code">
     <attribute name="title">
      <string>Code</string>
     </attribute>
     <widget class="QWidget" name="verticalLayoutWidget">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>901</width>
        <height>261</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_12">
         <item>
          <widget class="QLabel" name="FunctionReturn">
           <property name="maximumSize">
            <size>
             <width>160</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="text">
            <string>TextLabel</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="FunctionHeader">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximumSize">
            <size>
             <width>800</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="readOnly">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="FunctionParameters">
           <property name="text">
            <string>TextLabel</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="regenerate">
           <property name="text">
            <string>Regenerate</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>{</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_13">
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeType">
            <enum>QSizePolicy::Maximum</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPlainTextEdit" name="FunctionBody"/>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_15">
         <item>
          <spacer name="horizontalSpacer_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeType">
            <enum>QSizePolicy::Maximum</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLabel" name="FunctionBodyReturn">
           <property name="text">
            <string>FunctionBodyReturn</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>260</y>
        <width>899</width>
        <height>15</height>
       </rect>
      </property>
      <property name="text">
       <string>}</string>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>931</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="separator"/>
    <addaction name="actionLoad"/>
    <addaction name="actionSave"/>
    <addaction name="actionView_as_C_Source"/>
    <addaction name="actionSave_As_C_Source"/>
    <addaction name="separator"/>
    <addaction name="actionView_as_Arduino_Source"/>
    <addaction name="actionSave_as_Arduiono_Source"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
    <addaction name="separator"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <action name="actionLoad">
   <property name="text">
    <string>Load</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
  </action>
  <action name="actionSave_As_C_Source">
   <property name="text">
    <string>Save As C Source</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionView_as_C_Source">
   <property name="text">
    <string>View as C Source</string>
   </property>
  </action>
  <action name="actionNew">
   <property name="text">
    <string>New</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionView_as_Arduino_Source">
   <property name="text">
    <string>View as Arduino Source</string>
   </property>
  </action>
  <action name="actionSave_as_Arduiono_Source">
   <property name="text">
    <string>Save as Arduiono Source</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>GraphWidget</class>
   <extends>QGraphicsView</extends>
   <header location="global">graphwidget.h</header>
  </customwidget>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
